sdk/storage/Azure.Storage.DataMovement/src/Shared/DataTransferStatusInternal.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Storage.DataMovement
+{
+    internal class DataTransferStatusInternal : DataTransferStatus
```

Comment:
It looks like we would do this if we needed to construct a DataTransferStatus in our support packages like Azure.Storage.DataMovement.Blobs. But unless github isn't rendering the file that does it, I don't see where we do that. I'd rather just throw an internal constructor on the base type if we need it. This exact constructor but it just calls `: this()` instead of `:base()`.

Reply:
I made this and I hope my explanation won't be too confusing :). We can sync offline if it's confusing.

So we share the internal source files of the checkpointer / job plan schema classes between the base DataMovement package and the Blobs DataMovement package. However when I first attempted to call the `protected internal` constructor of the `DataTransferStatus(state, hasFailureItems, hasSkippedItems)`, it had trouble compiling in both packages since this was being called in the shared internal checkpointer / job plan schema classes.

My first thought was just to make the constructor `public` but I think it would have been unideal to allow someone to create a status like this publicly. I can get onboard with this as long as everyone else is okay with this.

Then my solution to get around it being public was to make this internal class.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/38637